cypress/e2e/SP_29_Course_Study_Hour_Estimator.cy.js

```diff
@@ -0,0 +1,71 @@
+describe('Test Course Study Hour Estimator [SP-29]', function() {
+
+    beforeEach(() => {
+        cy.visit(Cypress.env('baseUrl'));
+        cy.get('[data-test="email"]').type('test1@gmail.com');
+        cy.get('[data-test="password"]').type('test1');
+        cy.get('[data-test="login"]').click();
+    })
+
+    it('Add Course Soen 331', function() {
+        cy.get('[data-test="addEventButtonCalendarPage"]').click();
+        cy.get('[value="course"]').click();
+        cy.get('#subject').type('Soen');
+        cy.get('#catalog').type('331');
+        cy.get('#eventHeader').type('Intro Formal Methods for Software Engineering');
+        cy.get('.MuiFormControlLabel-root').click(); // click recurrent
+        cy.get('.MuiFormGroup-root > :nth-child(2)').click(); // click every week
+        cy.get('[data-test="recurrence-end-date"]').click();
+        cy.get('.MuiDialogActions-root > :nth-child(2)').click();
+        cy.get('#startTime').click().type("11:45:00");
+        cy.get('#endTime').click().type("13:00:00");
+        cy.get('[data-test="addButton"]').click();
+        cy.get('[data-test="Intro Formal Methods for Software Engineering"]').should('exist'); //inside typography of eventDisplay
+        cy.wait(2000);
+    })
+
+    it('Add Course Soen 385', function() {
```

Comment:
seems Sonar cloud doesnt like the repetition of the code, it labels it as duplicate code

Reply:
I will make some functions!